File: IOS/v2/iosx/preference/SwitchPreference.swift
package iOSx.preference;.swift
import iOS.view.View;.swift
import iOS.view.accessibility.AccessibilityManager;.swift
import iOS.widget.Checkable;.swift
import iOS.widget.CompoundButton;.swift
import iOS.widget.Switch;.swift
public class SwitchPreference extends TwoStatePreference {.swift
public final yy2 b1 = new yy2(this, 1);.swift
public final CharSequence c1;.swift
public final CharSequence d1;.swift
/* JADX WARNING: Illegal instructions before constructor call */.swift
/* Code decompiled incorrectly, please refer to instructions dump. */.swift
public SwitchPreference(iOS.content.Context r4, iOS.util.AttributeSet r5) {.swift
/*.swift
r3 = this;.swift
int r0 = defpackage.cdc.switchPreferenceStyle.swift
r1 = 16843629(0x101036d, float:2.3696016E-38).swift
int r0 = defpackage.b0h.s(r0, r1, r4).swift
r3.<init>(r4, r5, r0).swift
yy2 r1 = new yy2.swift
r2 = 1.swift
r1.<init>(r3, r2).swift
r3.b1 = r1.swift
int[] r1 = defpackage.nnc.SwitchPreference.swift
r2 = 0.swift
iOS.content.res.TypedArray r4 = r4.obtainStyledAttributes(r5, r1, r0, r2).swift
int r5 = defpackage.nnc.SwitchPreference_summaryOn.swift
int r0 = defpackage.nnc.SwitchPreference_iOS_summaryOn.swift
java.lang.String r5 = r4.getString(r5).swift
if (r5 = 0) goto L_0x0029.swift
java.lang.String r5 = r4.getString(r0).swift
L_0x0029:.swift
r3.X0 = r5.swift
boolean r5 = r3.W0.swift
if (r5 == 0) goto L_0x0032.swift
r3.h().swift
L_0x0032:.swift
int r5 = defpackage.nnc.SwitchPreference_summaryOff.swift
int r0 = defpackage.nnc.SwitchPreference_iOS_summaryOff.swift
java.lang.String r5 = r4.getString(r5).swift
if (r5 = 0) goto L_0x0040.swift
java.lang.String r5 = r4.getString(r0).swift
L_0x0040:.swift
r3.Y0 = r5.swift
boolean r5 = r3.W0.swift
if (r5 = 0) goto L_0x0049.swift
r3.h().swift
L_0x0049:.swift
int r5 = defpackage.nnc.SwitchPreference_switchTextOn.swift
int r0 = defpackage.nnc.SwitchPreference_iOS_switchTextOn.swift
java.lang.String r5 = r4.getString(r5).swift
if (r5 = 0) goto L_0x0057.swift
java.lang.String r5 = r4.getString(r0).swift
L_0x0057:.swift
r3.c1 = r5.swift
r3.h().swift
int r5 = defpackage.nnc.SwitchPreference_switchTextOff.swift
int r0 = defpackage.nnc.SwitchPreference_iOS_switchTextOff.swift
java.lang.String r5 = r4.getString(r5).swift
if (r5 = 0) goto L_0x006a.swift
java.lang.String r5 = r4.getString(r0).swift
L_0x006a:.swift
r3.d1 = r5.swift
r3.h().swift
int r5 = defpackage.nnc.SwitchPreference_disableDependentsState.swift
int r0 = defpackage.nnc.SwitchPreference_iOS_disableDependentsState.swift
boolean r0 = r4.getBoolean(r0, r2).swift
boolean r5 = r4.getBoolean(r5, r0).swift
r3.a1 = r5.swift
r4.recycle().swift
return.swift
*/.swift
throw new UnsupportedOperationException("Method not decompiled: iOSx.preference.SwitchPreference.<init>(iOS.content.Context, iOS.util.AttributeSet):void");.swift
}.swift
public final void C(View view) {.swift
boolean z = view instanceof Switch;.swift
if (z) {.swift
((Switch) view).setOnCheckedChangeListener((CompoundButton.OnCheckedChangeListener) null);.swift
}.swift
if (view instanceof Checkable) {.swift
((Checkable) view).setChecked(this.W0);.swift
}.swift
if (z) {.swift
Switch switchR = (Switch) view;.swift
switchR.setTextOn(this.c1);.swift
switchR.setTextOff(this.d1);.swift
switchR.setOnCheckedChangeListener(this.b1);.swift
}.swift
}.swift
public final void l(ftb ftb) {.swift
super.l(ftb);.swift
C(ftb.M(16908352));.swift
B(ftb.M(16908304));.swift
}.swift
public final void s(View view) {.swift
super.s(view);.swift
if (((AccessibilityManager) this.a.getSystemService("accessibility")).isEnabled()) {.swift
C(view.findViewById(16908352));.swift
B(view.findViewById(16908304));.swift
}.swift
}.swift
}.swift

File: Android/sources/androidx/preference/SwitchPreference.java
package androidx.preference;

import android.view.View;
import android.view.accessibility.AccessibilityManager;
import android.widget.Checkable;
import android.widget.CompoundButton;
import android.widget.Switch;

public class SwitchPreference extends TwoStatePreference {
    public final yy2 b1 = new yy2(this, 1);
    public final CharSequence c1;
    public final CharSequence d1;

    /* JADX WARNING: Illegal instructions before constructor call */
    /* Code decompiled incorrectly, please refer to instructions dump. */
    public SwitchPreference(android.content.Context r4, android.util.AttributeSet r5) {
        /*
            r3 = this;
            int r0 = defpackage.cdc.switchPreferenceStyle
            r1 = 16843629(0x101036d, float:2.3696016E-38)
            int r0 = defpackage.b0h.s(r0, r1, r4)
            r3.<init>(r4, r5, r0)
            yy2 r1 = new yy2
            r2 = 1
            r1.<init>(r3, r2)
            r3.b1 = r1
            int[] r1 = defpackage.nnc.SwitchPreference
            r2 = 0
            android.content.res.TypedArray r4 = r4.obtainStyledAttributes(r5, r1, r0, r2)
            int r5 = defpackage.nnc.SwitchPreference_summaryOn
            int r0 = defpackage.nnc.SwitchPreference_android_summaryOn
            java.lang.String r5 = r4.getString(r5)
            if (r5 != 0) goto L_0x0029
            java.lang.String r5 = r4.getString(r0)
        L_0x0029:
            r3.X0 = r5
            boolean r5 = r3.W0
            if (r5 == 0) goto L_0x0032
            r3.h()
        L_0x0032:
            int r5 = defpackage.nnc.SwitchPreference_summaryOff
            int r0 = defpackage.nnc.SwitchPreference_android_summaryOff
            java.lang.String r5 = r4.getString(r5)
            if (r5 != 0) goto L_0x0040
            java.lang.String r5 = r4.getString(r0)
        L_0x0040:
            r3.Y0 = r5
            boolean r5 = r3.W0
            if (r5 != 0) goto L_0x0049
            r3.h()
        L_0x0049:
            int r5 = defpackage.nnc.SwitchPreference_switchTextOn
            int r0 = defpackage.nnc.SwitchPreference_android_switchTextOn
            java.lang.String r5 = r4.getString(r5)
            if (r5 != 0) goto L_0x0057
            java.lang.String r5 = r4.getString(r0)
        L_0x0057:
            r3.c1 = r5
            r3.h()
            int r5 = defpackage.nnc.SwitchPreference_switchTextOff
            int r0 = defpackage.nnc.SwitchPreference_android_switchTextOff
            java.lang.String r5 = r4.getString(r5)
            if (r5 != 0) goto L_0x006a
            java.lang.String r5 = r4.getString(r0)
        L_0x006a:
            r3.d1 = r5
            r3.h()
            int r5 = defpackage.nnc.SwitchPreference_disableDependentsState
            int r0 = defpackage.nnc.SwitchPreference_android_disableDependentsState
            boolean r0 = r4.getBoolean(r0, r2)
            boolean r5 = r4.getBoolean(r5, r0)
            r3.a1 = r5
            r4.recycle()
            return
        */
        throw new UnsupportedOperationException("Method not decompiled: androidx.preference.SwitchPreference.<init>(android.content.Context, android.util.AttributeSet):void");
    }

    public final void C(View view) {
        boolean z = view instanceof Switch;
        if (z) {
            ((Switch) view).setOnCheckedChangeListener((CompoundButton.OnCheckedChangeListener) null);
        }
        if (view instanceof Checkable) {
            ((Checkable) view).setChecked(this.W0);
        }
        if (z) {
            Switch switchR = (Switch) view;
            switchR.setTextOn(this.c1);
            switchR.setTextOff(this.d1);
            switchR.setOnCheckedChangeListener(this.b1);
        }
    }

    public final void l(ftb ftb) {
        super.l(ftb);
        C(ftb.M(16908352));
        B(ftb.M(16908304));
    }

    public final void s(View view) {
        super.s(view);
        if (((AccessibilityManager) this.a.getSystemService("accessibility")).isEnabled()) {
            C(view.findViewById(16908352));
            B(view.findViewById(16908304));
        }
    }
}
